tripy/tripy/frontend/utils.py

```diff
@@ -84,11 +85,23 @@ def _add_column_info(arg, arg_index, is_kwarg, num_positional, func_name, skip_n
 
     # Find the first caller of this function that is NOT the function registry.
     # Also save the last dispatch target we see.
+
+    # Start from the registry. It will always be present except for tests,
```

Comment:
One more place where it could be missing is methods of classes. If that becomes an issue, then maybe we could update the function registry to automatically discover and wrap methods too.

Reply:
Would there be an example of a class method for which we would use the decorator? I thought we were only using it for overrides of magic methods.

Reply:
Since it also handles conversion to shapes now, there could be cases in the future where we want to use it on a method. Could you add a note in the docstring of the decorator mentioning the assumptions it makes about the call stack?

Reply:
Hopefully it won't be necessary if we remove these hacks per the above discussion, will note whatever assumptions ultimately remain.